# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../mapnik/mapnik-0.7.1-force-boost-filesystem-v2.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Force the use of boost filesystem v2

Mapnik 0.7.1 needs boost filesystem v2, but boost 1.46 is using v3 by
default. Until a new release of Mapnik is done which is using boost
filesystem v3 we enforce the use of boost filesystem v2. 

Origin: http://trac.mapnik.org/changeset/2496

--- mapnik-0.7.1/SConstruct.orig	2010-03-22 17:40:47.000000000 +0100
+++ mapnik-0.7.1/SConstruct	2011-05-17 10:46:30.592002343 +0200
@@ -1015,7 +1015,9 @@
                 env.Append(CXXFLAGS = common_cxx_flags + '-O %s' % ndebug_flags)
         else:
             # Common flags for GCC.
-            gcc_cxx_flags = '-ansi -Wall %s -ftemplate-depth-100 %s' % (pthread, common_cxx_flags)
+            # note: starting with boost 1.46 boost filesystem v3 is default so force to be v2
+            # by using BOOST_FILESYSTEM_VERSION=2 (until we upgrade)
+            gcc_cxx_flags = '-ansi -Wall %s -ftemplate-depth-200 -DBOOST_FILESYSTEM_VERSION=2 %s' % (pthread, common_cxx_flags)
         
             if env['DEBUG']:
                 env.Append(CXXFLAGS = gcc_cxx_flags + '-O0 -fno-inline %s' % debug_flags)
